### PR TITLE
Fixes #19760 - Adds support for failover DHCP

### PR DESCRIPTION
**Pull request by sean797**
http://steronius.blogspot.co.uk/2015/10/dhcp-failover-on-rhel-7.html is worth reading if you would like.

Today the user would have to find their own way of replicating leases - I'm hoping in the future Foreman can create the leases on both proxies but that would require some changes to Core.

### CODE REVIEW

File: manifests/proxydhcp.pp [hunk 1]
```diff
@@ -24,6 +24,12 @@
     $nameservers = split($foreman_proxy::dhcp_nameservers,',')
   }
 
+  if ( $foreman_proxy::dhcp_node_type =~ /^(primary|secondary)$/ ) {
```
Comment (ekohl) [inline manifests/proxydhcp.pp]: The braces aren't needed.

File: manifests/proxydhcp.pp [hunk 2]
```diff
@@ -40,5 +46,21 @@
     range          => $foreman_proxy::dhcp_range,
     gateway        => $foreman_proxy::dhcp_gateway,
     search_domains => $foreman_proxy::dhcp_search_domains,
+    failover       => $failover,
+  }
+
+  if ( $foreman_proxy::dhcp_node_type =~ /^(primary|secondary)$/ ) {
```
Comment (ekohl) [inline manifests/proxydhcp.pp]: You could also write ```if $failover {```

**Comment by sean797** [comment]
Thanks @ekohl 👍 Updated commit, filed a new ticket because I cannot move the old one & tests passed

**Comment by mmoll** [comment]
yes 💥
merged, thanks @sean797!

**Pull request merged**